docs/versioning.md

```diff
@@ -0,0 +1,136 @@
+# Electron Versioning Document
+
+## Version 1.x
+
+Electron versions *< 2.0* have not conformed to the [semver](http://semver.org) spec. Major versions corresponded to end-user API changes. Minor versions corresponded to Chromium major releases. Patch versions corresponded to new features and bug fixes. While convenient for developers merging features, it creates problems for developers of client-facing applications. The QA testing cycles of major apps like Slack, Stride, Teams, Skype, VS Code, Atom, and Desktop can be lengthy and stability is a highly desired outcome. There is a high risk in adopting new features while trying to absorb bug fixes.
```

Comment:
This first paragraph dives into some nitty gritty that is not going to be applicable to most folks reading this document. I would suggest leading with our recommended best practices for versioning, and move these historical details farther down in the doc.

Reply:
I like having a short-but-sweet 1.x explanation at the beginning. I imagine 1.x may take a while to disappear, but I'd eventually be into moving this to the bottom once it's truly historic.